packages/mds-db/schema.ts

```diff
@@ -337,25 +337,25 @@ const COLUMN_TYPE: { [C in COLUMN_NAME]: string } = {
   [COLUMN.audit_subject_id]: 'varchar(255) NOT NULL',
   [COLUMN.audit_trip_id]: 'uuid NOT NULL',
   [COLUMN.avg_violation_dist]: 'double precision',
-  [COLUMN.bad_events]: 'json',
+  [COLUMN.bad_events]: 'json NOT NULL',
   [COLUMN.base_url]: 'varchar(127) NOT NULL',
-  [COLUMN.bin_size]: 'varchar(31)',
+  [COLUMN.bin_size]: 'varchar(31) NOT NULL',
   [COLUMN.capacity]: 'jsonb',
   [COLUMN.charge]: 'real',
   [COLUMN.cross_street]: 'varchar(255)',
   [COLUMN.deleted]: 'bigint',
   [COLUMN.description]: 'varchar(255)',
   [COLUMN.device_id]: 'uuid NOT NULL',
   [COLUMN.distance]: 'double precision',
-  [COLUMN.duration]: 'bigint',
+  [COLUMN.duration]: 'bigint NOT NULL',
   [COLUMN.effective_date]: 'bigint',
   [COLUMN.end_service_area_id]: 'uuid',
-  [COLUMN.end_time]: 'bigint',
-  [COLUMN.event_counts]: 'json',
-  [COLUMN.event_time_violations]: 'json',
+  [COLUMN.end_time]: 'bigint NOT NULL',
+  [COLUMN.event_counts]: 'json NOT NULL',
+  [COLUMN.event_time_violations]: 'json NOT NULL',
   [COLUMN.event_type]: 'varchar(31)',
   [COLUMN.event_type_reason]: 'varchar(31)',
-  [COLUMN.events]: 'json[]',
+  [COLUMN.events]: 'json[] NOT NULL',
```

Comment:
json[] is interesting ... isn't an array of json just json?

Reply:
Is that true? I'm not too familiar with postgres data types but this has been working for me... I do want to investigate moving to jsonb types, they're binary forms that allow for more efficient storage and fetching so I'll play around with this one.